specs/IsPointInDraggableRegion.md

```diff
@@ -0,0 +1,303 @@
+                ncrbuttondown_on_draggable_region_ = TRUE
+                SetCapture(hwnd); // capture the mouse
+                return 0;
+            }
+```
+#### .NET / WinRT
+```c#
+        if (m.msg == WM_NCRBUTTONDOWN) {
+            Point p = new Point(GET_X_LPARAM(m.lParam), GET_Y_LPARAM(m.lParam));
+
+            if(webview2->IsPointInDraggableRegion(ref p) // this clause is not required but recommended, see above for more details
+            && m.wParam == HTCAPTION) {
+                setCapture(hwnd);
+                ncrbuttondown_on_draggable_region_ = true;
+                m.Result = 0;
+            }
+        }
+```
+
+### Handle WM_RBUTTONUP
+Capturing the mouse earlier allows us to receive this event. We need to verify two things; the point is within a draggable region, and the previous R button-down message was within a draggable region (using ncrbuttondown_on_draggable_region_ ). 
+
+If both clauses are true, then you may proceed to handle this as you, please. Most likely you would want to show the system menu.
+#### Win32
+```cpp
+        case WM_RBUTTONUP:
+            BOOL isInRegion = FALSE;
+            webview2->IsPointInDraggableRegion(&pt, &isInRegion);
+
+            if(isInRegion && ncrbuttondown_on_draggable_region_ == TRUE) {
+                // show system menu
+                POINT screen_point{ GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam) };
+
+                MapWindowPoints(hwnd, HWND_DESKTOP, &screen_point, 1);
+                
+                UINT flags =
+                    TPM_LEFTALIGN | TPM_TOPALIGN | TPM_RIGHTBUTTON | TPM_RETURNCMD;
+                
+                HMENU menu = GetSystemMenu(hwnd, FALSE);
+
+                const int command = TrackPopupMenu(
+                    menu, flags, screen_point.x, screen_point.y, 0, hwnd, nullptr);
+
+                if (command)
+                    SendMessage(hwnd, WM_SYSCOMMAND, static_cast<WPARAM>(command),
+                            0);
+            
+            }
+            
+        break;
+```
+#### .NET / WinRT
+```c#
+        if (m.msg == WM_RBUTTONUP) {
+
+            if (webview2.IsPointInDraggableRegion(ref pt) // this clause is not required but recommended, see below for more details
+            && ncrbuttondown_on_draggable_region_ == true)
+            {
+                // show system menu
+                POINT screen_point = new POINT(GET_X_LPARAM(lParam), GET_Y_LPARAM(lParam));
+
+                MapWindowPoints(hwnd, HWND_DESKTOP, ref screen_point, 1);
+
+                uint flags = TPM_LEFTALIGN | TPM_TOPALIGN | TPM_RIGHTBUTTON | TPM_RETURNCMD;
+
+                IntPtr menu = GetSystemMenu(hwnd, false);
+
+                int command = TrackPopupMenu(menu, flags, screen_point.x, screen_point.y, 0, hwnd, IntPtr.Zero);
+
+                if (command != 0)
+                {
+                    SendMessage(hwnd, WM_SYSCOMMAND, (IntPtr)command, IntPtr.Zero);
+                }
+            
+            }
+        }
+```
```

Comment:
It seems like the different chunks of code are all part of one scenario. They share members and depend on the sections above them. If that's correct, it will probably be easier to understand if they are stated that way - as one scenario with a COM section and then a .NET/WinRT section.